File: app/src/main/java/ua/regin/wallpapers/utils/ResourceUtils.java
package ua.regin.wallpapers.utils;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.support.annotation.DrawableRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public class ResourceUtils {
    @Nullable
    @SuppressWarnings("deprecation")
    public static Drawable getDrawable(@NonNull Context context, @DrawableRes int drawableId) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
            return context.getDrawable(drawableId);
        return context.getResources().getDrawable(drawableId);
    }
}

File: app/src/main/java/ua/regin/wallpapers/util/ResourceUtils.kt
package ua.regin.wallpapers.util;

import android.content.Context
import android.graphics.drawable.Drawable
import android.os.Build

public class ResourceUtils {

    @Suppress("DEPRECATION")
    public fun getDrawable(context: Context, drawableId: Int): Drawable {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
            return context.getDrawable(drawableId);
        return context.resources.getDrawable(drawableId);
    }

    companion object Helper {
        fun create(): ResourceUtils = ResourceUtils()
    }
}